php/php.editor/src/org/netbeans/modules/php/editor/verification/WrongOrderOfArgsHint.java

```diff
@@ -150,12 +150,18 @@ public String getDescription() {
         @Override
         public void implement() throws Exception {
             EditList edits = new EditList(doc);
-            OffsetRange offsetRange = getOffsetRange();
-            StringBuilder sb = new StringBuilder();
-            for (ParameterElement param : functionDeclarationInfo.getParameters()) {
-                sb.append(param.asString(OutputType.COMPLETE_DECLARATION)).append(", "); //NOI18N
+            List<FormalParameter> originalParameters = node.getFormalParameters();
+            List<ParameterElement> parameters = functionDeclarationInfo.getParameters();
+            assert originalParameters.size() == parameters.size();
+            // maybe, in the case of constructor property promotion
+            // parameters can be declared with multiple lines
+            // so, replace the rearranged parameters with positions of original parameters
+            // instead of replacing whole parameters as one line
+            for (int i = 0; i < originalParameters.size(); i++) {
+                FormalParameter originalParameter = originalParameters.get(i);
+                OffsetRange originalRange = new OffsetRange(originalParameter.getStartOffset(), originalParameter.getEndOffset());
+                edits.replace(originalRange.getStart(), originalRange.getLength(), parameters.get(i).asString(OutputType.COMPLETE_DECLARATION), false, 0);
```

Comment:
for the following case:
```php
class Example {
    public function __construct(
        private string $x = "default",
        $y,
    ) {}
}

```
In this case, I think that users don't prefer to be recreated the parameters as one line. So, I've changed the behavior.

Reply:
Yes, this is more user friendly.
If user spends time splitting parameters on separate line then it is better to keep the formatting after reordering parameters.